templates/templates/meganav/base.html

```diff
@@ -203,92 +212,117 @@
-              tabindex = "-1"
-            %}
-              {% include "templates/meganav/_list-item.html" %}
-            {% endwith %}
-          {% endfor %}
-        {% endfor %}
-        {% if side_nav_section.secondary_links %}
-          {% for links_section in side_nav_section.secondary_links %}
-          <ul class="p-list p-navigation__secondary-links">
-              <p class="p-muted-heading is-muted">{{ links_section.title }}</p>
-              {% for link in links_section.links %}
-                <li class="p-list__item">
-                  <a class="p-navigation__secondary-link" href="{{ link.url }}" tabindex="-1">{{ link.title }}</a>
-                </li>
-              {% endfor %}
-            </ul>
+      <li class="p-navigation__item--dropdown-toggle"
+          id="{{ id }}-{{ format_to_id(side_nav_section.title) }}"
+          data-id="{{ id }}-{{ format_to_id(side_nav_section.title) }}">
+        <button class="p-navigation__link"
+                href="#{{ id }}-{{ format_to_id(side_nav_section.title) }}-content-mobile"
+                aria-controls="{{ id }}-{{ format_to_id(side_nav_section.title) }}-content-mobile"
+                role="tab"
+                tabindex="-1">{{ side_nav_section.title }}</button>
+        <ul class="p-navigation__dropdown"
+            id="{{ id }}-{{ format_to_id(side_nav_section.title) }}-content-mobile"
+            aria-hidden="true"
+            data-level="2">
+          <li class="p-navigation__item--dropdown-close"
+              id="{{ id }}-{{ format_to_id(side_nav_section.title) }}-back">
+            <button class="p-navigation__link js-back"
+                    href="{{ id }}-{{ format_to_id(side_nav_section.title) }}-content-mobile"
+                    aria-controls="{{ id }}-{{ format_to_id(side_nav_section.title) }}-content-mobile"
+                    tabindex="-1">Back</button>
+          </li>
+          {% for links_section in side_nav_section.primary_links %}
+            {% if links_section.title %}
+              <li class="p-navigation__dropdown-item--heading">
+                <h2 class="p-muted-heading">{{ links_section.title }}</h2>
+              </li>
+            {% endif %}
+            {% for link in links_section.links %}
+              {% with
+                url = link.url,
+                title = link.title,
+                description = link.description,
+                secondary_cta_url = link.secondary_cta_url,
+                secondary_cta_title = link.secondary_cta_title,
+                tabindex = "-1"
+                %}
+                {% include "templates/meganav/_list-item.html" %}
+              {% endwith %}
+            {% endfor %}
           {% endfor %}
-        {% endif %}
-      </ul>
-    </li>
+          {% if side_nav_section.secondary_links %}
+            {% for links_section in side_nav_section.secondary_links %}
+              <li class="p-navigation__dropdown-item">
```

Comment:
The `p-navigation__dropdown-item` (confusingly) is meant for the link inside the dropdown - it has styles for hover effects etc.

Unfortunately it seems we currently don't have any style for dropdown items that are not links (don't have hover but have correct paddings). Which makes things harder.

It also seems that in Vanilla dropdowns `<li>` elements in them don't have any class at all, making it even harder to properly style.

But overall it seem that to keep this closer to Vanilla we would need to not use `p-navigation__dropdown-item` on li elements, but have some other class name for them.

Reply:
Good question, I'm not entirely sure myself.

The point is, as per Vanilla the `p-navigation__dropdown-item` should not be on `li`, just on the link inside.
So, if there is a link (or anything else?) inside `li` class should be moved there.

But, we don't have anything in Vanilla for dropdown items that are not clickable (like headers or quick link block), so I guess this will still require some custom class/CSS.

It may be easier to discuss on a call maybe.